### PR TITLE
Interns/multiroot sfdx

### PR DESCRIPTION
**Pull request by cmatthiassonsf**

### CODE REVIEW

File: packages/lightning-lsp-common/src/shared.ts [hunk 1]
```diff
@@ -45,17 +45,18 @@ export function getSfdxProjectFile(root: string) {
...
+    //     return detectWorkspaceHelper(workspaceRoots[0]);
+    // }
+    // for (const root of workspaceRoots) {
+    //     const type = detectWorkspaceHelper(root);
+    //     if (type !== WorkspaceType.CORE_PARTIAL) {
+    //         console.error('unknown workspace type');
+    //         return WorkspaceType.UNKNOWN;
+    //     }
+    // }
+    // return WorkspaceType.CORE_PARTIAL;
```
Comment (nrkruk) [inline packages/lightning-lsp-common/src/shared.ts]: Is this no longer needed?

File: packages/lightning-lsp-common/src/context.ts [hunk 2]
```diff
@@ -38,15 +38,26 @@ export class WorkspaceContext {
     public indexers: Map<string, Indexer> = new Map();
 
     private findNamespaceRootsUsingTypeCache: () => Promise<{ lwc: string[]; aura: string[] }>;
-    private initSfdxProjectConfigCache: () => Promise<ISfdxProjectConfig>;
+    // private initSfdxProjectConfigCache: () => Promise<ISfdxProjectConfig>;
```
Comment (nrkruk) [inline packages/lightning-lsp-common/src/context.ts]: Delete

File: packages/lightning-lsp-common/src/context.ts [hunk 3]
```diff
@@ -38,15 +38,26 @@ export class WorkspaceContext {
...
+        if (typeof workspaceRoots === 'string') {
+            this.workspaceRoots = [path.resolve(workspaceRoots)];
+        } else {
+            const resolved: string[] = [];
+            for (const ws of workspaceRoots) {
+                resolved.push(path.resolve(ws));
+            }
+            this.workspaceRoots = resolved;
+        }
+        // this.workspaceRoots = typeof workspaceRoots === 'string' ? [path.resolve(workspaceRoots)] : workspaceRoots;
```
Comment (nrkruk) [inline packages/lightning-lsp-common/src/context.ts]: Delete

File: packages/lwc-language-server/src/metadata-utils/__tests__/util.ts [hunk 4]
```diff
@@ -17,14 +17,23 @@ export async function validate(
...
     await indexer(context, true);
 }
+// const context = new class TestContext extends WorkspaceContext {
+//     public getSfdxProjectConfig(): Promise<ISfdxProjectConfig[]> {
+//         return Promise.resolve({
+//             sfdxProjectConfig : ISfdxProjectConfig[]
+//         });
+//     }
+// }(join(process.cwd(), 'test-workspaces', testWorkspace));
+// await context.configureProject();
```
Comment (nrkruk) [inline packages/lwc-language-server/src/metadata-utils/__tests__/util.ts]: Is this no longer needed? If so lets delete it

File: packages/lwc-language-server/src/metadata-utils/content-assets-util.ts [hunk 5]
```diff
@@ -46,18 +49,22 @@ function processContentAssets(workspace: string, writeConfig: boolean): Promise<
...
     const { workspaceRoots } = context;
-    const { sfdxPackageDirsPattern } = await context.getSfdxProjectConfig();
-    const CONTENT_ASSET_GLOB_PATTERN = `${sfdxPackageDirsPattern}/**/contentassets/*.asset-meta.xml`;
-
-    try {
-        const files: string[] = await glob(CONTENT_ASSET_GLOB_PATTERN, { cwd: workspaceRoots[0] });
-        for (const file of files) {
-            CONTENT_ASSETS.add(getResourceName(file));
+    // const { sfdxPackageDirsPattern } = await context.getSfdxProjectConfig();
+    // const CONTENT_ASSET_GLOB_PATTERN = `${sfdxPackageDirsPattern}/**/contentassets/*.asset-meta.xml`;
```
Comment (nrkruk) [inline packages/lwc-language-server/src/metadata-utils/content-assets-util.ts]: Delete

File: packages/lwc-language-server/src/metadata-utils/custom-labels-util.ts [hunk 6]
```diff
@@ -50,17 +50,22 @@ export function resetCustomLabels() {
...
 export async function indexCustomLabels(context: WorkspaceContext, writeConfigs: boolean = true): Promise<void> {
     const { workspaceRoots } = context;
-    const { sfdxPackageDirsPattern } = await context.getSfdxProjectConfig();
-    const CUSTOM_LABEL_GLOB_PATTERN = `${sfdxPackageDirsPattern}/**/labels/CustomLabels.labels-meta.xml`;
-    try {
-        const files: string[] = await glob(CUSTOM_LABEL_GLOB_PATTERN, { cwd: workspaceRoots[0] });
-        for (const file of files) {
-            CUSTOM_LABEL_FILES.add(join(workspaceRoots[0], file));
+    // const { sfdxPackageDirsPattern } = await context.getSfdxProjectConfig();
+    // const CUSTOM_LABEL_GLOB_PATTERN = `${sfdxPackageDirsPattern}/**/labels/CustomLabels.labels-meta.xml`;
```
Comment (nrkruk) [inline packages/lwc-language-server/src/metadata-utils/custom-labels-util.ts]: Delete

File: packages/lwc-language-server/src/metadata-utils/static-resources-util.ts [hunk 7]
```diff
@@ -33,7 +33,10 @@ export async function updateStaticResourceIndex(updatedFiles: FileEvent[], { wor
         }
     }
     if (didChange) {
-        return processStaticResources(workspaceRoots[0], writeConfigs);
+        for (const ws of workspaceRoots) {
+            processStaticResources(ws, writeConfigs);
+        }
+        return;
```
Comment (nrkruk) [inline packages/lwc-language-server/src/metadata-utils/static-resources-util.ts]: Does processStaticResources return anything?

File: packages/lwc-language-server/src/metadata-utils/static-resources-util.ts [hunk 8]
```diff
@@ -45,17 +48,22 @@ async function processStaticResources(workspace: string, writeConfigs: boolean):
...
 export async function indexStaticResources(context: WorkspaceContext, writeConfigs: boolean = true): Promise<void> {
     const { workspaceRoots } = context;
-    const { sfdxPackageDirsPattern } = await context.getSfdxProjectConfig();
-    const STATIC_RESOURCE_GLOB_PATTERN = `${sfdxPackageDirsPattern}/**/staticresources/*.resource-meta.xml`;
-    try {
-        const files: string[] = await glob(STATIC_RESOURCE_GLOB_PATTERN, { cwd: workspaceRoots[0] });
-        for (const file of files) {
-            STATIC_RESOURCES.add(getResourceName(file));
+    // const { sfdxPackageDirsPattern } = await context.getSfdxProjectConfig();
+    // const STATIC_RESOURCE_GLOB_PATTERN = `${sfdxPackageDirsPattern}/**/staticresources/*.resource-meta.xml`;
```
Comment (nrkruk) [inline packages/lwc-language-server/src/metadata-utils/static-resources-util.ts]: Delete

File: packages/lwc-language-server/src/metadata-utils/static-resources-util.ts (hunk 7 above)
Comment (nrkruk) [reply packages/lwc-language-server/src/metadata-utils/static-resources-util.ts]: This used to return the result from processStaticResources(), so just making sure callers of this function aren't expecting something in the return value

File: test-workspaces/core-like-workspace/app/main/core/ui-force-components/.vscode/settings.json [hunk 9]
```diff
@@ -0,0 +1,40 @@
+{
```
Comment (nrkruk) [inline test-workspaces/core-like-workspace/app/main/core/ui-force-components/.vscode/settings.json]: delete this file

File: test-workspaces/sfdx-multi-workspace/sfdx-workspace-1/.sfdx/typings/lwc/apex.d.ts [hunk 10]
```diff
@@ -0,0 +1,31 @@
+declare module "@salesforce/apex" {
```
Comment (nrkruk) [inline test-workspaces/sfdx-multi-workspace/sfdx-workspace-1/.sfdx/typings/lwc/apex.d.ts]: Probably should delete this entire .sfdx/typings/ directory to ensure it gets created each time during tests

File: test-workspaces/sfdx-multi-workspace/sfdx-workspace-1/force-app/main/default/lwc/todo/store.js [hunk 11]
```diff
@@ -0,0 +1,9 @@
+const STORAGE_KEY = 'todos';
```
Comment (nrkruk) [inline test-workspaces/sfdx-multi-workspace/sfdx-workspace-1/force-app/main/default/lwc/todo/store.js]: Do we run automated tests on each one of these components? If not we should probably delete some of them

File: test-workspaces/sfdx-multi-workspace/sfdx-workspace-2/.sfdx/typings/lwc/contentassets.d.ts [hunk 12]
```diff
@@ -0,0 +1,4 @@
+declare module "@salesforce/contentAssetUrl/logo" {
```
Comment (nrkruk) [inline test-workspaces/sfdx-multi-workspace/sfdx-workspace-2/.sfdx/typings/lwc/contentassets.d.ts]: Again, I'd delete these .sfdx/typings directories in each test workspace

File: packages/lwc-language-server/src/metadata-utils/static-resources-util.ts (hunk 7 above)
Comment (cmatthiassonsf) [reply packages/lwc-language-server/src/metadata-utils/static-resources-util.ts]: processStaticResources does a fs.writeFIle so no real return value. empty return in updateStaticResourceIndex should be ok.

**Pull request closed without merging**